mesonbuild/compilers/compilers.py

```diff
@@ -133,11 +133,15 @@ def is_header(fname: 'mesonlib.FileOrString') -> bool:
 def is_source_suffix(suffix: str) -> bool:
     return suffix in source_suffixes
 
+@lru_cache(maxsize=None)
+def cached_is_source_by_name(fname: 'mesonlib.FileOrString') -> bool:
+    suffix = fname.split('.')[-1].lower()
```

Comment:
This annotation looks broken as the entire point of the additional function is to implement caching on `str` source name for `is_source()`.

Reply:
Also wrong in the existing `cached_by_name`, I'll fix both.